src/sentry/event_manager.py

```diff
@@ -819,10 +823,13 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
         # attempt to find a matching hash
         all_hashes = self._find_hashes(project, hashes)
 
-        try:
-            existing_group_id = six.next(h.group_id for h in all_hashes if h.group_id is not None)
-        except StopIteration:
-            existing_group_id = None
+        existing_group_id = None
+        for h in all_hashes:
+            if h.group_id is not None:
+                existing_group_id = h.group_id
+                break
+            if h.group_tombstone_id is not None:
+                raise HashDiscarded('Matches discarded group %s' % h.group_tombstone_id)
```

Comment:
This exception message is a little confusing since it says "group" but then references the tombstone ID